### PR TITLE
Swagger for canceling MI management operations

### PR DESCRIPTION
**Pull request by toki95**
Adding swagger for cancel and list of management operations on managed server.

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+            "Cancelled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ManagementOperationState",
+            "modelAsString": true
+          }
+        },
+        "errorCode": {
```
Comment (yungezz) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: is there min and max?

Comment (toki95) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: I am not sure that I understand what you mean.

Comment (toki95) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: Also can you add other people for ARM team, I don't have permission to do that. Jared is on vacation until the end of the year.

**Review by akromm-zz — APPROVED** [review]

**Comment by toki95** [comment]
@yungezz can you please merge this pull request?

**Comment by yungezz** [comment]
hi @toki95 , this PR is adding new version, need ARM review and signoff.

**Comment by toki95** [comment]
@yungezz what does that mean? Adam Krom  has signed off and he is from ARM team since Jared if oof until end of the year.

**Comment by yungezz** [comment]
hi @akromm  could you pls label the PR with "ARMSignedOff"?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json [hunk 2]
```diff
@@ -0,0 +1,358 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-06-01-preview",
```
Comment (pilor) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: For future PRs please copy the previous version's swagger spec in commit 1 before making changes in future commits.  This makes it much easier to review.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        "operationId": "ManagedInstanceOperations_Cancel",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagedInstanceNameParameter"
+          },
+          {
+            "name": "operationId",
```
Comment (pilor) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: This should be "operationName".  "ID" typically refers to the full resource ID and the list of operations gives the value you want the user to provider here in the "name" property

File: specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/operations/{operationId}/cancel": {
```
Comment (pilor) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: You need a GET API for individual operations to be compliant with the RPC especially if you are now allowing people to cancel individual operations.  A comparison is operations on a template deployment:  https://docs.microsoft.com/en-us/rest/api/resources/deploymentoperations/get

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json (hunk 4 above)
Comment (toki95) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: This API for cancel was done in a manner to be compliant with SQL DB and they don't have an API fro getting one particular operation. 
Also can you please explain if there is a requirement to create a new API or if we can add not-required parameter in existing API for listing all operations-operationId which will allow customer to get one particular operation.

Comment (pilor) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: Existing APIs not following [the RPC ](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#nested-resources) shouldn't be a reason to continue that trend.  Services within Azure rely on RPs following the RPC so they can work with resources in a generic way without needing to know the quirks of any specific API.

I have not seen an example in existing swagger specs where someone modeled a LIST and GET operation in the same operation via an optional name parameter.  The response model would be different so I don't imagine that is a good idea.

**Comment by yungezz** [comment]
pls address ARM reviewing comments.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json (hunk 4 above)
Comment (toki95) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/ManagedInstanceOperations.json]: I don't think that we understood each other entirely. What I was thinking of doing is adding new key in existing API for getting all operations.
Current call of API:
https://management.azure.com/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/v-urmila/providers/Microsoft.Sql/managedInstances/v-urmila-mi-test/operations?api-version=2018-06-01-preview
which returns all operations to customer.
And adding a key _operationId_ the same API can be called like this:
https://management.azure.com/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/v-urmila/providers/Microsoft.Sql/managedInstances/v-urmila-mi-test/operations/a64077e8-43e6-47b6-a6ef-9c2f40fac38b?api-version=2018-06-01-preview

**Comment by yungezz** [comment]
hi @toki95   pls address reviewing comments .

**Comment by yungezz** [comment]
hi @toki95 pls address reviewing comments .

**Comment by yungezz** [comment]
hi @toki95 any update? thanks.

**Comment by yungezz** [comment]
ping again. @toki95

**Pull request closed without merging**